src/PrestaShopBundle/EventListener/Admin/Context/LanguageContextListener.php

```diff
@@ -28,22 +28,53 @@
 
 namespace PrestaShopBundle\EventListener\Admin\Context;
 
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
 use PrestaShop\PrestaShop\Core\Context\EmployeeContext;
 use PrestaShop\PrestaShop\Core\Context\LanguageContextBuilder;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
 
 /**
  * Listener dedicated to set up Language context for the Back-Office/Admin application.
  */
-class LanguageContextListener
+class LanguageContextListener implements EventSubscriberInterface
 {
+    /**
+     * Priority lower than EmployeeContextListener so that EmployeeContext is correctly initialized
+     */
+    public const KERNEL_REQUEST_PRIORITY = EmployeeContextListener::KERNEL_REQUEST_PRIORITY - 1;
+
     public function __construct(
         private readonly LanguageContextBuilder $languageContextBuilder,
         private readonly EmployeeContext $employeeContext,
+        private readonly ConfigurationInterface $configuration,
     ) {
     }
 
-    public function onKernelRequest(RequestEvent $event): void
+    public static function getSubscribedEvents()
+    {
+        return [
+            KernelEvents::REQUEST => [
+                // Before RouterListener
+                ['initDefaultLanguageContext', 33],
```

Comment:
It would be interesting to add 33 as a constant

Reply:
@M0rgan01 I just added it, but since I couldn't think of a good place that would store this common constant I added the same constant in both listeners 😅 If you have a better suggestion I'm all for it